lib/components/primitive-components/Trace/Trace_doInitialPcbManualTraceRender.ts

```diff
@@ -70,11 +70,21 @@ export function Trace_doInitialPcbManualTraceRender(trace: Trace) {
     layer: layer as LayerRef,
     start_pcb_port_id: anchorPort.pcb_port_id!,
   })
+  // Get group rotation in radians
+  const group = trace.getGroup()
+  const rotation = (group?._parsedProps.pcbRotation || 0) * (Math.PI / 180)
```

Comment:
what happens if a group is rotated inside another group that has rotation? This will fail.

You can use the `group._getGlobalTransform()` to get a transformation-matrix that can be applied to all points (I may have gotten the name slightly wrong)

Use the `transformation-matrix` library to applied the rotation to each point. It should have a function that handles this.

This will also handle translations etc.